package/base-files/files/lib/functions/ipv4.sh

```diff
@@ -0,0 +1,185 @@
+uint_max=4294967295
+
+# Common error handler
+__die() {
+    local fmt="$1"
+    shift
+
+    printf "$fmt\n" "$@" >&2
+    exit 1
+}
+
+# check that $1 is only base 10 digits, and that it doesn't
+# exceed 2^32-1
+assert_uint32() {
+    local n="$1"
+
+    if [ -z "$n" -o -n "${n//[0-9]/}" ]; then
+	printf "Not a decimal integer (%s)" "$n ">&2
+	return 1
+    fi
+
+    if [ "$n" -gt $uint_max ]; then
+	printf "Out of range (%s)" "$n" >&2
+	return 1
+    fi
+
+    if [ "$((n + 0))" != "$n" ]; then
+	echo "Not normalized notation (%s)" "$n" >& 2
```

Comment:
echo/printf mixup + typo at the end of the line (space between `& 2`).

Reply:
The space doesn't make a difference but I changed it anyway.